fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/AccountNumberGenerator.java

```diff
@@ -143,8 +166,13 @@ private String generateAccountNumber(Map<String, String> propertyMap, AccountNum
             }
             if (accountNumberPrefixType.getValue().equals(AccountNumberPrefixType.PREFIX_SHORT_NAME.getValue())) {
                 Integer prefixLength = prefix.length();
-                Integer numberLength = accountMaxLength - prefixLength;
-                accountNumber = StringUtils.leftPad(propertyMap.get(ID), numberLength, '0');
+
+                if (randomAccountNumber.isEnabled()) {
+                    accountNumber = accountNumber.substring(prefixLength);
```

Comment:
How do we ensure the substring is unique? What if we had generated a string that only differs from existing ones by the positions that are now being removed? Would it not make more sense to do the full generation of account ID (including any prefix etc) and only after that check for duplicates? Or some other similar way to ensure uniqueness...